### PR TITLE
Added ProxyConfig interface and implemented in VelocityConfiguration

### PR DESCRIPTION
**Pull request by Crypnotic**
This adds the preliminary code to have access to the proxy configuration values through the API. Picked up from #77 after inactivity for some time. No javadocs as of yet

### CODE REVIEW

File: proxy/src/main/java/com/velocitypowered/proxy/connection/client/StatusSessionHandler.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public boolean handle(StatusRequest packet) {
                 new ServerPing.Version(shownVersion, "Velocity " + ProtocolConstants.SUPPORTED_GENERIC_VERSION_STRING),
                 new ServerPing.Players(server.getPlayerCount(), configuration.getShowMaxPlayers(), ImmutableList.of()),
                 configuration.getMotdComponent(),
-                configuration.getFavicon(),
+                configuration.getFavicon().get(),
```
[the remark this answers is not in the document]
Comment (Crypnotic) [reply proxy/src/main/java/com/velocitypowered/proxy/connection/client/StatusSessionHandler.java]: I wrote that with the assumption that the Favicon was either set or null. Is that not the case? If it is, does that need to be changed to make sure nothing weird happens?

File: proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java [hunk 2]
```diff
@@ -263,8 +264,8 @@ public int getLoginRatelimit() {
         return advanced.getLoginRatelimit();
     }
 
-    public Favicon getFavicon() {
-        return favicon;
+    public Optional<Favicon> getFavicon() {
+        return Optional.of(favicon);
```
Comment (astei) [inline proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java]: `Optional.ofNullable()`

Comment (astei) [inline proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java]: `Optional.ofNullable()`

File: api/src/main/java/com/velocitypowered/api/proxy/config/ProxyConfig.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package com.velocitypowered.api.proxy.config;
+
+import com.velocitypowered.api.util.Favicon;
+import net.kyori.text.Component;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public interface ProxyConfig {
```
Comment (astei) [inline api/src/main/java/com/velocitypowered/api/proxy/config/ProxyConfig.java]: Make sure to add in Javadoc.

Comment (Crypnotic) [reply api/src/main/java/com/velocitypowered/api/proxy/config/ProxyConfig.java]: I went ahead and wrote javadoc for a methods, not the best so please feel free to make them not suck so much

**Pull request merged**